instance/models/openedx_instance.py

```diff
@@ -227,7 +227,10 @@ def appserver_is_shut_down(appserver):
             """
             if appserver.status == AppServerStatus.Terminated:
                 return True
-            configuration_failed = appserver.status == AppServerStatus.ConfigurationFailed
+            configuration_failed = (
+                appserver.status == AppServerStatus.ConfigurationFailed or
+                appserver.status == AppServerStatus.Error
+            )
             vm_terminated = appserver.server.status == ServerStatus.Terminated
             return configuration_failed and vm_terminated
```

Comment:
@bdero @pomegranited You're right, to determine whether a single `appserver_is_shut_down`, we could just check whether the corresponding VM has been terminated, and stop taking the status of the app server itself into account. I think the reason we're not doing that is that we (initially) wanted to make sure that the UI would only hide an instance if we can be sure it has been "properly" shutdown. If an instance had one or more app servers that got stuck, e.g., in `ConfiguringServer`, we wanted to be aware of it, to be able to investigate.

> I feel like I'm more confused about the difference between `AppServerStatus.Terminated` and `ServerStatus.Terminated` than anything here.

* `AppServerStatus.Terminated` indicates that an app server was running successfully at some point (i.e., it never entered `AppServerStatus.ConfigurationFailed` or `AppServerStatus.Error`) and has since been shut down, cf. https://github.com/open-craft/opencraft/blob/master/instance/models/appserver.py#L111.
* `ServerStatus.Terminated` indicates that a VM has been terminated/shut down, cf. https://github.com/open-craft/opencraft/blob/master/instance/models/server.py#L98. It might have been in any of the available [server states](https://github.com/open-craft/opencraft/blob/master/instance/models/server.py#L74) before that.